xhttp/serve.go

```diff
@@ -41,11 +81,12 @@ func Serve(ctx context.Context, shutdownTimeout time.Duration, s *http.Server, l
 		shutdownCtx, cancel := context.WithTimeout(xcontext.WithoutCancel(ctx), shutdownTimeout)
 		defer cancel()
 
-		err := s.Shutdown(shutdownCtx)
+		err := ss.Shutdown(shutdownCtx)
 		<-serverClosed // Wait for server to exit
```

Comment:
the previous changes were not reverted

Reply:
@alixander can you just do a reset on the master branch ? 

tho i really don't understand why a revert is necessary here.